scripts/publish/build.js

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env node
+'use strict';
+
+/*eslint-disable no-console */
+const denodeify = require('denodeify');
+const fs = require('fs');
+const glob = denodeify(require('glob'));
+const path = require('path');
+const npmRun = require('npm-run');
+const rimraf = denodeify(require('rimraf'));
+
+
+const root = path.join(__dirname, '../..');
+const dist = path.join(root, 'dist');
+const packagesRoot = path.join(root, 'packages');
+
+function copy(from, to) {
+  from = path.relative(process.cwd(), from);
+  to = path.relative(process.cwd(), to);
+  return new Promise((resolve, reject) => {
+    const rs = fs.createReadStream(from);
+    const ws = fs.createWriteStream(to);
+
+    rs.on('error', reject);
+    ws.on('error', reject);
+    ws.on('close', resolve);
+
+    rs.pipe(ws);
+  });
+}
+
+
+function patch() {
+  const filePath = path.join(__dirname, '../../node_modules/@types/common-tags/common-tags.d.ts');
+  const content = fs.readFileSync(filePath, 'utf8')
+    .replace(/literals: string\[\]/, 'literals: TemplateStringsArray');
+  fs.writeFileSync(filePath, content, 'utf8');
+}
+
+// First delete the dist folder.
+Promise.resolve()
+  .then(() => console.log('Deleting dist folder...'))
+  .then(() => rimraf(dist))
+  .then(() => console.log('Patching node_modules typings...'))
+  .then(() => patch())
+  .then(() => console.log('Compiling packages...'))
+  .then(() => {
+    const packages = require('../../lib/packages');
+    return Object.keys(packages)
+      .reduce((promise, packageName) => {
+        const pkg = packages[packageName];
+        const name = path.relative(packagesRoot, pkg.root);
+
+        return promise.then(() => {
+          console.log(`  ${name}`);
+          return npmRun.execSync(`tsc -p ${path.relative(process.cwd(), pkg.root)}`);
+        });
+      }, Promise.resolve());
+  })
+  .then(() => console.log('Copying uncompiled resources...'))
+  .then(() => glob(path.join(packagesRoot, '**/*')))
+  .then(files => {
+    console.log(`  Found ${files.length} files...`);
+    return files
+      .map((fileName) => path.relative(packagesRoot, fileName))
+      .filter((fileName) => {
+        if (/^angular-cli.blueprints/.test(fileName)) {
```

Comment:
Shouldn't this be `/^angular-cli\/blueprints/` instead? There doesn't seem to be any file with `angular-cli.blueprints` in it's name.